lib/logstash/plugin_mixins/event_support/event_factory_adapter.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: utf-8
+
+require 'logstash/plugin_mixins/event_support'
+
+require 'logstash-core'
+require 'logstash/plugin'
+
+module LogStash
+  module PluginMixins
+    module EventSupport
+      module EventFactoryAdapter
+
+        ##
+        # @api internal
+        # @param base [Class]: a class that inherits `LogStash::Plugin`, typically one
+        #                      descending from one of the four plugin base classes
+        #                      (e.g., `LogStash::Inputs::Base`)
+        # @raise [ArgumentError]
+        # @return [void]
+        def self.included(base)
+          fail(ArgumentError, "`#{base}` must inherit LogStash::Plugin") unless base < LogStash::Plugin
+
+          if defined? LogStash::Plugins::EventFactorySupport
+            core_event_support = LogStash::Plugins::EventFactorySupport
+            fail(ArgumentError, "`#{base}` should include #{core_event_support}") unless base < core_event_support
+          else
+            require 'logstash/plugin_mixins/event_support/event_factory_adapter/fallback_impl'
```

Comment:
Nit: Since Logstash has such a long include path and our many plugins aren't always the best at properly namespacing their files, I prefer `require_relative` for library-internal requires.

```suggestion
            require_relative 'event_factory_adapter/fallback_impl'
```